File: m2-h2/src/main/java/ru/yegorpilipenko/otus/spring/m2h2/controller/AuthorController.java
package ru.yegorpilipenko.otus.spring.m2h2.controller;

import lombok.RequiredArgsConstructor;
import org.springframework.shell.standard.ShellComponent;
import org.springframework.shell.standard.ShellMethod;
import ru.yegorpilipenko.otus.spring.m2h2.document.Author;
import ru.yegorpilipenko.otus.spring.m2h2.service.AuthorService;

import javax.annotation.Nonnull;
import java.util.Collection;

import static java.lang.String.format;
import static java.util.stream.Collectors.toList;

@ShellComponent
@RequiredArgsConstructor
public final class AuthorController extends AbstractShellController {

    @Nonnull
    private final AuthorService authorService;

    @Nonnull
    @ShellMethod(key = "authors", value = "Show all authors.", group = "Author Commands")
    public Collection<String> getAll() {
        return authorService.getAll().stream().map(this::toView).collect(toList());
    }

    @Nonnull
    private String toView(@Nonnull Author author) {
        return format("%s %s", author.getFirstName(), author.getLastName());
    }

}

File: m1-h6/src/main/kotlin/ru/yegorpilipenko/otus/spring/m1h6/controller/AuthorController.kt
package ru.yegorpilipenko.otus.spring.m1h6.controller

import org.springframework.shell.standard.ShellComponent
import org.springframework.shell.standard.ShellMethod
import ru.yegorpilipenko.otus.spring.m1h6.entity.Author
import ru.yegorpilipenko.otus.spring.m1h6.service.AuthorService

@ShellComponent
class AuthorController(val authorService: AuthorService) {

    @ShellMethod(value = "Show all authors.", group = "Author Commands")
    fun authors(): Collection<String>
            = authorService.findAll().map { author -> "${author.id}: ${author.firstName} ${author.lastName}" }

    @ShellMethod(value = "Find author by ID.", group = "Author Commands")
    fun author(id: Long): String {
        val author = authorService.findById(id)
        return if (author != null) "${author.id}: ${author.firstName} ${author.lastName}" else "Not found"
    }

    @ShellMethod(value = "Save author.", group = "Author Commands")
    fun authorSave(firstName: String, lastName: String): String {
        val author = authorService.save(Author(firstName = firstName, lastName = lastName))
        return "${author.id}: ${author.firstName} ${author.lastName}"
    }

    @ShellMethod(value = "Remove author by ID.", group = "Author Commands")
    fun authorRemove(id: Long): String = if (authorService.removeById(id)) "Removed" else "Not found"

}
